src/main/java/ruslan/shastkiv/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -44,4 +47,26 @@ public ResponseEntity<Map<String, String>> handleRegistrationException(
 
         return new ResponseEntity<>(errorResponse, HttpStatus.CONFLICT);
     }
+
+    @ExceptionHandler(AuthenticationException.class)
+    public ResponseEntity<Map<String, String>> handleAuthenticationException(
+            AuthenticationException exception) {
+        Map<String, String> errorResponse = new HashMap<>();
+        errorResponse.put("error", exception.getMessage());
+        return new ResponseEntity<>(errorResponse, HttpStatus.UNAUTHORIZED);
+    }
+
+    @ExceptionHandler(BadCredentialsException.class)
+    public ResponseEntity<Map<String, String>> handleBadCredentialsException() {
+        Map<String, String> errorResponse = new HashMap<>();
+        errorResponse.put("error", "User password incorrect");
+        return new ResponseEntity<>(errorResponse, HttpStatus.UNAUTHORIZED);
+    }
+
+    @ExceptionHandler(JwtException.class)
```

Comment:
JwtException won't be handled by GlobalExceptionHandler because it appears in filter, so you can avoid to handle it or find the way to handle correctly